source/sn_coap_protocol.c

```diff
@@ -2439,22 +2455,20 @@ static bool sn_coap_protocol_update_duplicate_package_data(const struct coap_s *
                                                         const int16_t data_size,
                                                         const uint8_t *dst_packet_data_ptr)
 {
-    if (coap_msg_ptr->msg_type == COAP_MSG_TYPE_ACKNOWLEDGEMENT &&
-        handle->sn_coap_duplication_buffer_size != 0) {
-        coap_duplication_info_s* info = sn_coap_protocol_linked_list_duplication_info_search(handle,
-                                                                                             dst_addr_ptr,
-                                                                                             coap_msg_ptr->msg_id);
-
-        /* Update package data to duplication info struct if it's not there yet */
-        if (info && info->packet_ptr == NULL) {
-            info->packet_ptr = handle->sn_coap_protocol_malloc(data_size);
-            if (info->packet_ptr) {
-                memcpy(info->packet_ptr, dst_packet_data_ptr, data_size);
-                info->packet_len = data_size;
-            } else {
-                tr_error("sn_coap_protocol_update_duplication_package_data - failed to allocate duplication info!");
-                return false;
-            }
+    coap_duplication_info_s* info = sn_coap_protocol_linked_list_duplication_info_search(handle,
+                                                                                         dst_addr_ptr,
+                                                                                         coap_msg_ptr->msg_id);
+
+    /* Update package data to duplication info struct if it's not there yet */
+    if (info && info->packet_ptr == NULL) {
```

Comment:
This change is now affecting all callers sn_coap_protocol_update_duplicate_package_data() and there were two of them, most notably sn_coap_protocol_build(). I'm a bit worried it will increase overall memory consumption now everywhere.

Reply:
To be on the safe side, I added the suggested new implementation for this case. Code-wise it adds 240 bytes of flash. RAM impact seems negligible in either case, but would most likely require must more extensive testing.

Reply:
Why not just do the splitting, keep the if() block you removed in original then call this from it?

Reply:
Optimized the implementation as suggested. The impact is now only 24 bytes.